tests/rptest/tests/datalake/datalake_services.py

```diff
@@ -0,0 +1,143 @@
+                             },
+                             *args,
+                             **kwargs)
+        self.test_ctx = test_ctx
+        self.filesystem_catalog_mode = filesystem_catalog_mode
+        self.spark: Optional[SparkService] = None
+        self.trino: Optional[TrinoService] = None
+
+    def setUp(self):
+        self.catalog_service.set_filesystem_wrapper_mode(
+            self.filesystem_catalog_mode)
+        super().setUp()
+        si = self.redpanda.si_settings
+        self.spark = SparkService(
+            self.test_ctx,
+            iceberg_catalog_rest_uri=str(self.catalog_service.catalog_url),
+            cloud_storage_access_key=str(si.cloud_storage_access_key),
+            cloud_storage_secret_key=str(si.cloud_storage_secret_key),
+            cloud_storage_region=si.cloud_storage_region,
+            cloud_storage_api_endpoint=str(si.endpoint_url))
+        self.spark.start()
+        self.trino = TrinoService(
+            self.test_ctx,
+            iceberg_catalog_rest_uri=str(self.catalog_service.catalog_url),
+            cloud_storage_access_key=str(si.cloud_storage_access_key),
+            cloud_storage_secret_key=str(si.cloud_storage_secret_key),
+            cloud_storage_region=si.cloud_storage_region,
+            cloud_storage_api_endpoint=str(si.endpoint_url))
+        self.trino.start()
+
+    def tearDown(self):
+        if self.spark:
+            self.spark.stop()
+        if self.trino:
+            self.trino.stop()
+        return super().tearDown()
+
+    def create_iceberg_enabled_topic(self,
+                                     name,
+                                     partitions=1,
+                                     replicas=1,
+                                     translation_interval_ms=3000):
+        rpk = RpkTool(self.redpanda)
+        rpk.create_topic(topic=name,
+                         partitions=partitions,
+                         replicas=replicas,
+                         config={
+                             TopicSpec.PROPERTY_ICEBERG_ENABLED:
+                             "true",
+                             TopicSpec.PROPERTY_ICEBERG_TRANSLATION_INTERVAL:
+                             translation_interval_ms
+                         })
+
+    def wait_for_iceberg_table(self, schema, table, timeout, backoff_sec):
+        client = self.catalog_service.client("redpanda-iceberg-catalog")
+
+        def table_created():
+            namespaces = client.list_namespaces()
+            self.logger.debug(f"namespaces: {namespaces}")
+            return (schema, ) in namespaces and (
+                schema, table) in client.list_tables(schema)
+
+        wait_until(
+            table_created,
+            timeout_sec=timeout,
+            backoff_sec=backoff_sec,
+            err_msg=
+            f"Timed out waiting {schema}.{table} to be created in the catalog")
+
+    def wait_for_events(self, topic, msg_count, timeout=30, backoff_sec=5):
```

Comment:
Just curious where the "event" terminology is coming from? I would've thought SQL engines would refer to rows and such

Reply:
Yea in my mind, this file is focused on validating a datalake, which uses the same nomenclature a database. Anyway, it's fine as is